0000-route-broadcasting-protocol/0000-route-broadcasting-protocol.md

```diff
@@ -0,0 +1,226 @@
+- `Route Update Request`
+  - A request that conveys routing information and ask the receiver node to update its routing table based on the information.
+
+#### Nodes
+- `Node A`
+  - Connected to `Node B`.
+  - A node that wants routing information from `Node B` to build its routing table.
+  - `Node A` wants `Node B` to send route update requests that convey routing information.
+- `Node B`
+  - Connected to `Node A`.
+  - A node whose status is IDLE at this point that means this node doesn't send any route update request to `Node A`.
+
+#### Procedure
+This is a normal procedure for instance.
+
+- `Node A` sends a **route control request** of `SYNC` to `Node B`
+- `Node B` receives the route control request and updates its status to `SYNC`
+  - This means that from hence, `Node B` sends route update requests to provide routing information for `Node A`
+- `Node B` responds that the state is correctly updated to `Node A`
+- (some time after the procedure above)
+- `Node B` sends a **route update request** that has routing information (some logs) to `Node A`
+- `Node A` receives the route update request, and updates its routing table based on the provided information.
+- `Node A` responds that the routing table is correctly updated to `Node B`
+
+#### Packet
+The request and the response above are transferred in [ILP packets](../0027-interledger-protocol-4/0027-interledger-protocol-4.md#specification). 
+
+The `fulfillment` of the response packet is always a zero-filled 32 byte octet string, therefor the condition is always the SHA-256 hash digest of that, i.e. the Base64 decoded value of `Zmh6rfhivXdsj8GLjp+OIAiXFIVu4jOzkCpZHQ1fKSU=`.
+
+As with other bilateral protocols the packets are addressed directly to the peer using the `peer.` address prefix. RBP packets are specifically identified using the address `peer.route`.
+
+All current implementations of RBP default to an amount of `0` in the ILP packets used to request route control or update.
+
+### Route Control
+#### Procedure
+Route control is done in the following procedure:
+
+- A node requests IDLE or SYNC mode to the corresponded node.
+- The corresponded node changes its status if needed, and respond that the request is done.
+- If the receiving node is not configured to send route updates, the receiving node responds with an error.
+- If the request cannot be deserialized or interpreted as appropriate, the corresponded node responds with an error.
+
+#### Packet
+- Request
+  - The `type` of the ILP packet is `ILP Prepare` (type id: 12)
+  - The `amount` of the ILP packet is `0`
+  - The `expiresAt` of the ILP packet is arbitrary
+  - The `executionCondition` of the ILP packet is `Zmh6rfhivXdsj8GLjp+OIAiXFIVu4jOzkCpZHQ1fKSU=` in Base64 format
+  - The `destination` address of the ILP packet is `peer.route.control`
+  - The `data` of the ILP packet is described at `RouteControlRequestData` in [ASN.1 definition](#asn1-definition)
```

Comment:
For forward compatibility, nodes should ignore any extra data that they do not understand.